### PR TITLE
Henry/board unit test

### PR DESCRIPTION
**Pull request by henrytran28**
🚁 🐰 💯

### CODE REVIEW

File: src/board/board_test.lua [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        ghostSquares = board.ghostTetromino.squares
+
+        assert.are_equal(4, ghostSquares[1].x)
+        assert.are_equal(0, ghostSquares[1].y)
+        assert.are_equal(5, ghostSquares[2].x)
+        assert.are_equal(0, ghostSquares[2].y)
+        assert.are_equal(5, ghostSquares[3].x)
+        assert.are_equal(1, ghostSquares[3].y)
+        assert.are_equal(4, ghostSquares[4].x)
+        assert.are_equal(1, ghostSquares[4].y)
```
Comment (ctlai95) [inline src/board/board_test.lua]: I'd recommend testing for ghost movement as well

**Comment by henrytran28** [comment]
@ctlai95 Please re-review senpai

File: src/board/board_test.lua [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        board.gravityFrameCounter.maxFrames = 5
+        currentTetrominoID = board.currentTetromino.id
+        board:handleGravity(10)
+
+        -- gravity timer should have reset and moved down one
+        movedDownTetromino = Tetromino:new(currentTetrominoID,
+                properties.SPAWN[currentTetrominoID], colors.ASH)
+        movedDownTetromino:offset(0, -1)
+        assert.are_equal(0, board.gravityFrameCounter.elapsedFrames)
+        assert.are.same(movedDownTetromino.origin, board.currentTetromino.origin)
```
Comment (ctlai95) [inline src/board/board_test.lua]: This just checks that currentTetromino moved down right?
I would just do something like 
```suggestion
        assert.are.equal(18, board.currentTetromino.origin.y)
```
and instantiate it at height 19 above

File: src/board/board_test.lua [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        -- add squares in the way and test if there is an obstacle below
+        board.currentTetromino = board.nextTetromino
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(3, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(4, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(5, 0), colors.ASH))
+        table.insert(board.boardTetrominoSquares, Square:new(Point:new(4, 1), colors.ASH))
+
+        board:updateMatrices()
+        board.currentTetromino:offset(0, -17)
+        board:obstacleBelow(board.currentTetromino.squares)
```
Comment (ctlai95) [inline src/board/board_test.lua]: I think you forgot an assert here haha

File: src/board/board_test.lua [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        assert.are_equal(1, ghostSquares[1].y)
+        assert.are_equal(8, ghostSquares[2].x)
+        assert.are_equal(1, ghostSquares[2].y)
+        assert.are_equal(7, ghostSquares[3].x)
+        assert.are_equal(1, ghostSquares[3].y)
+        assert.are_equal(9, ghostSquares[4].x)
+        assert.are_equal(0, ghostSquares[4].y)
+    end)
+
+    it("CycleNextTetromino", function()
```
Comment (ctlai95) [inline src/board/board_test.lua]: let's add some `assert.is_truthy` for currentTetromino, ghostTetromino, and nextTetromino

**Comment by henrytran28** [comment]
So I can't remove some places because it requires stuff to be in the currentTetromino.squares... 😅

File: src/board/board_test.lua [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    it("UpdateMatrices", function()
+        for i = 0, board.width - 1, 1 do
+            for j = 0, board.height - 1, 1 do
+                assert.are.equal(0, board.boardTetrominosMatrix[i][j])
+            end
+        end
+
+        table.insert(board.boardTetrominoSquares,
+        Square:new(Point:new(0, 0), colors.ASH))
+            board:updateMatrices()
```
[the remark this answers is not in the document]
Comment (henrytran28) [reply src/board/board_test.lua]: There is no rule on line continuation but I thought we can just stick to one indent.

[the remark this answers is not in the document]
Comment (henrytran28) [reply src/board/board_test.lua]: So there is no rule on line continuation its just whatever you want it to be. I was thinking of just doing one indent but I can just make it really long.

**Comment by henrytran28** [comment]
So I can't remove some of them because the matrix and the squares are decoupled... 😅

**Comment by ctlai95** [comment]
> So I can't remove some of them because the updateMatrices are in some of the board function and it clears the matrix... 😅

Ok thats fine then, please disregard 😋

**Comment by henrytran28** [comment]
> > So I can't remove some of them because the updateMatrices are in some of the board function and it clears the matrix... 😅
> 
> Ok thats fine then, please disregard 😋

I can change some though.

File: src/board/board_test.lua (hunk 5 above)
[the remark this answers is not in the document]
Comment (henrytran28) [reply src/board/board_test.lua]: I'm actually blind

**Pull request merged**